src/pull_module/libgit2.cpp

```diff
@@ -200,17 +202,85 @@ HfDownloader::HfDownloader() {
     this->hfEndpoint = "";
     this->hfToken = "";
     this->httpProxy = "";
+    this->overwritePath = false;
+}
+
+std::string HfDownloader::getGraphDirectory() {
+    return this->downloadPath;
+}
+
+std::string HfDownloader::getGraphDirectory(const std::string& inDownloadPath, const std::string& inSourceModel) {
+    std::string fullPath = FileSystem::joinPath({inDownloadPath, inSourceModel});
+    // Windows path creation
+    if (FileSystem::getOsSeparator() != "/") {
+        std::replace(fullPath.begin(), fullPath.end(), '/', '\\');
+    }
+
+    return fullPath;
+}
+
+HfDownloader::HfDownloader(const std::string& inSourceModel, const std::string& inDownloadPath, const std::string& inHfEndpoint, const std::string& inHfToken, const std::string& inHttpProxy, bool inOverwrite) {
+    this->sourceModel = inSourceModel;
+    this->downloadPath = HfDownloader::getGraphDirectory(inDownloadPath, inSourceModel);
+    this->hfEndpoint = inHfEndpoint;
+    this->hfToken = inHfToken;
+    this->httpProxy = inHttpProxy;
+    this->overwritePath = inOverwrite;
 }
 
-HfDownloader::HfDownloader(const std::string& sourceModel, const std::string& downloadPath, const std::string& hfEndpoint, const std::string& hfToken, const std::string& httpProxy) {
-    this->sourceModel = sourceModel;
-    this->downloadPath = downloadPath;
-    this->hfEndpoint = hfEndpoint;
-    this->hfToken = hfToken;
-    this->httpProxy = httpProxy;
+Status HfDownloader::RemoveReadonlyFileAttributeFromDir(const std::string& directoryPath) {
```

Comment:
We could do that only for files that we create not on path just passed onto us from user.